src/engine.py

```
@@ -508,6 +510,35 @@ def __init__(self, config: AppConfig):
+        if task.cancelled():
+            return
+        exc = task.exception()
+        if exc:
+            logger.error(
+                "Background task failed",
+                task_name=task.get_name(),
+                error=str(exc),
+                exc_info=exc,
+            )
+
+    def _fire_and_forget(self, coro, *, name: str = None) -> asyncio.Task:
```

Ruff (0.15.0)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)